src/screens/ProfileScreen/elements/ProfilePicture.js

```diff
@@ -24,36 +27,55 @@ const ProfilePicture = ({
   return (
     <View style={styles.wrapImageProfile}>
       <TouchableNativeFeedback onPress={onImageContainerClick}>
-        <CircleGradient fill={karmaScore} size={normalize(100)} width={normalize(6)}>
-          <View style={styles.profileImageContainer}>
+        {withKarma ? (
+          <CircleGradient
+            fill={karmaScore}
+            size={normalize(size)}
+            width={normalize(width)}
+            testId="images">
+            <View style={styles.profileImageContainer(size)}>
+              <Image
+                testId="images"
+                style={styles.profileImage(size, width)}
+                source={{
+                  uri: profilePicPath ? `${profilePicPath}` : DEFAULT_PROFILE_PIC_PATH
+                }}
+                resizeMode={FastImage.resizeMode.stretch}
+              />
+              {renderAddIcon()}
+            </View>
+          </CircleGradient>
+        ) : (
+          <View style={styles.profileImageContainer(size)} testId="images">
             <Image
-              style={styles.profileImage}
+              testId="images"
+              style={styles.profileImage(size, width)}
               source={{
                 uri: profilePicPath ? `${profilePicPath}` : DEFAULT_PROFILE_PIC_PATH
               }}
               resizeMode={FastImage.resizeMode.stretch}
             />
             {renderAddIcon()}
           </View>
-        </CircleGradient>
+        )}
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The code for rendering the `CircleGradient` and the `View` components is almost identical, except for the wrapping `CircleGradient` component when `withKarma` is true. This repetition can be avoided by using a single render function that conditionally wraps the content with `CircleGradient` based on the `withKarma` prop. Here's how you could refactor this:

```diff
- {withKarma ? (
-   <CircleGradient
-     fill={karmaScore}
-     size={normalize(size)}
-     width={normalize(width)}
-     testId="images">
-     <View style={styles.profileImageContainer(size)}>
-       <Image
-         testId="images"
-         style={styles.profileImage(size, width)}
-         source={{
-           uri: profilePicPath ? `${profilePicPath}` : DEFAULT_PROFILE_PIC_PATH
-         }}
-         resizeMode={FastImage.resizeMode.stretch}
-       />
-       {renderAddIcon()}
-     </View>
-   </CircleGradient>
- ) : (
-   <View style={styles.profileImageContainer(size)} testId="images">
-      <Image
-       testId="images"
-       style={styles.profileImage(size, width)}
-        source={{
-          uri: profilePicPath ? `${profilePicPath}` : DEFAULT_PROFILE_PIC_PATH
-        }}
-        resizeMode={FastImage.resizeMode.stretch}
-      />
-      {renderAddIcon()}
-    </View>
- )}
+ {renderProfileContent(withKarma)}
```

And then define the `renderProfileContent` function as follows:

```javascript
const renderProfileContent = (withKarma) => {
  const content = (
    <View style={styles.profileImageContainer(size)} testId="images">
      <Image
        testId="images"
        style={styles.profileImage(size, width)}
        source={{
          uri: profilePicPath ? `${profilePicPath}` : DEFAULT_PROFILE_PIC_PATH
        }}
        resizeMode={FastImage.resizeMode.stretch}
      />
      {renderAddIcon()}
    </View>
  );

  return withKarma ? (
    <CircleGradient
      fill={karmaScore}
      size={normalize(size)}
      width={normalize(width)}
      testId="images">
      {content}
    </CircleGradient>
  ) : content;
};
```

This way, you avoid duplicating the code for rendering the image and the add icon.



<!-- This is an auto-generated comment by OSS CodeRabbit -->